__tests__/pages/p/[id].test.jsx

```diff
@@ -27,7 +40,7 @@ describe('Page', () => {
 
   it('renders a title', async () => {
   
-    global.prisma.post.findUnique.mockResolvedValue(post)
+    global.prisma.post.findUniqueOrThrow.mockResolvedValue(post)
```

Comment:
There was a type error coming from `getServerSideProps`. It is saying that whatever `getServerSideProps` returns should not return null. With `findUnique` it does return null if post is not found. But, with `findUniqueOrThrow`, it will always return post.